tox_pdm/plugin4.py

```diff
@@ -0,0 +1,232 @@
+
+from tox_pdm.utils import clone_pdm_files, get_env_lib_path, is_same_path, is_sub_array
+
+if t.TYPE_CHECKING:
+
+    from argparse import ArgumentParser
+
+    from tox.execute.api import ExecuteInstance, ExecuteOptions
+    from tox.execute.request import ExecuteRequest
+    from tox.execute.stream import SyncWrite
+    from tox.tox_env.api import ToxEnvCreateArgs
+    from tox.tox_env.register import ToxEnvRegister
+
+
+@impl
+def tox_add_option(parser: ArgumentParser) -> None:
+    os.environ["TOX_TESTENV_PASSENV"] = "PYTHONPATH"
+    parser.add_argument("--pdm", default="pdm", help="The executable path of PDM")
+
+
+@impl
+def tox_register_tox_env(register: ToxEnvRegister) -> t.Optional[bool]:
+    register.add_run_env(PdmRunner)
+    register.add_package_env(PdmPackageEnv)
+    register.default_run_env = "pdm"
+
+
+class Pdm(Python):
+    def __init__(self, create_args: ToxEnvCreateArgs) -> None:
+        self._executor: t.Optional[Execute] = None
+        self._installer: t.Optional[Pip] = None
+        self._package_path: t.Optional[Path] = None
+        super().__init__(create_args)
+
+    @property
+    def executor(self) -> Execute:
+        if not self._executor:
+            self._executor = PDMRunExecutor(self.options.is_colored)
+        return self._executor
+
+    @property
+    def installer(self) -> Pip:
+        if self._installer is None:
+            self._installer = PipPep582(self)
+        return self._installer
+
+    @property
+    def package_path(self) -> Path:
+        if not self._package_path:
+            self._package_path = Path(
+                get_env_lib_path(self.options.pdm, self.env_dir)
+            ).parent
+        return self._package_path
+
+    @property
+    def runs_on_platform(self) -> str:
+        return sys.platform
+
+    def create_python_env(self) -> None:
+        clone_pdm_files(self.env_dir, self.core["toxinidir"])
+        self.execute(
+            ["pdm", "use", "-f", self.base_python.extra["executable"]], StdinSource.OFF
+        )
+
+    def _get_python(self, base_python: t.List[str]) -> t.Optional[PythonInfo]:
+        interpreter = next(
+            filter(None, (get_interpreter(p, []) for p in base_python)), None
+        )
+        if not interpreter:
+            return None
+        return PythonInfo(
+            implementation=interpreter.implementation,
+            version_info=interpreter.version_info,
+            version=interpreter.version,
+            is_64=(interpreter.architecture == 64),
+            platform=interpreter.platform,
+            extra={"executable": Path(interpreter.system_executable)},
+        )
+
+    def python_cache(self) -> t.Dict[str, t.Any]:
+        base = super().python_cache()
+        base.update({"executable": str(self.base_python.extra["executable"])})
+        return base
+
+    def env_bin_dir(self) -> Path:
+        return self.package_path / ("Scripts" if os.name == "nt" else "bin")
```

Comment:
Is this hard coded for PDM? Shouldn't you here follow what sysconfig says instead?